inference-engine/tests/functional/inference_engine/conditional_compilation/selective_build.cpp

```diff
@@ -105,6 +112,51 @@ TEST(ConditionalCompilationTests, SwitchCase) {
 #undef CCTests_TestTemplateClass_cases
 }
 
+TEST(ConditionalCompilationTests, SwitchCaseWithDefault) {
+    // Default case is enabled
+#define CCTests_TestTemplateClass 1
+#define CCTests_TestTemplateClass_cases OV_DEFAULT()
+
+    int n = 0;
+
+    OV_SWITCH(CCTests, TestTemplateClass, n, 0,
+    OV_CASE(0, int),
+    OV_CASE(1, bool),
+    OV_CASE(2, float),
+    OV_DEFAULT());
+    EXPECT_EQ(n, 100);
+
+    n = 0;
+    EXPECT_EQ(n, 0);
+    OV_SWITCH(CCTests, TestTemplateClass, n, 1,
+    OV_CASE(0, int),
+    OV_CASE(1, bool),
+    OV_CASE(2, float),
+    OV_DEFAULT());
+    EXPECT_EQ(n, 100);
+
+    n = 0;
+    EXPECT_EQ(n, 0);
+    OV_SWITCH(CCTests, TestTemplateClass, n, 2,
+    OV_CASE(0, int),
+    OV_CASE(1, bool),
+    OV_CASE(2, float),
+    OV_DEFAULT());
+    EXPECT_EQ(n, 100);
+
+    n = 0;
+    EXPECT_EQ(n, 0);
+    OV_SWITCH(CCTests, TestTemplateClass, n, 3,
+    OV_CASE(0, int),
+    OV_CASE(1, bool),
+    OV_CASE(2, float),
+    OV_DEFAULT());
+    EXPECT_EQ(n, 100);
+
```

Comment:
Could you please add test with default case in different order like this:
    OV_SWITCH(CCTests, TestTemplateClass, n, 2,
    OV_CASE(0, int),
    OV_DEFAULT(bool),
    OV_CASE(2, float));

Reply:
At the current moment OV_DEFAULT should be the last

Reply:
If it should be the last we should do something which prevent writing wrong expressions but now my test seems compiling.